test/python/transpiler/test_coupling.py

```diff
@@ -448,6 +450,46 @@ def test_implements_iter(self):
         expected = [(0, 1), (1, 0), (1, 2), (2, 1)]
         self.assertEqual(sorted(coupling), expected)
 
+    def test_disjoint_coupling_map(self):
+        cmap = CouplingMap([[0, 1], [1, 0], [2, 3], [3, 2]])
+        self.assertFalse(cmap.is_connected())
+        distance_matrix = cmap.distance_matrix
+        expected = np.array(
+            [
+                [0, 1, 0, 0],
+                [1, 0, 0, 0],
+                [0, 0, 0, 1],
+                [0, 0, 1, 0],
```

Comment:
This is existing behavior I didn't want to change it here because that's a larger change outside the scope here. rustworkx's [`distance_matrix()`](https://qiskit.org/documentation/rustworkx/apiref/rustworkx.distance_matrix.html#rustworkx-distance-matrix) function has a `null_value` kwarg that we can set to `NaN` or `inf` to do this, but we should do that in a separate PR and document it with an `upgrade` note as potentially breaking for people.

Reply:
Or alternatively adding an optional argument to specify the null value in the return

Reply:
I'm not sure I follow how this could be existing behavior if it wouldn't have been possible to create a `CouplingMap` with unreachable nodes before this PR.

I could see a handful of cases where this could lead to confusion. e.g If I want to write code to cost a swap path for a qubit from index `i` to `j`, if `i==j`, I'd need zero. If it's impossible to reach `j` from `i`, I probably have a different problem.

If the return `dtype` is already `float`, it seems low-cost to use `math.nan` or `math.inf`. Other than using the individual component `CouplingMap` s, I don't see a way around passes needing to do _some_ additional check to know if the destination qubit is reachable.

Reply:
Ah, that's my bad I was thinking strictly in terms the `distance_matrix()` defaults but forgot that we were erroring if there coupling map wasn't connected previously and that we set `as_undirected=True` (so we never have issues with the reverse edges. So we'd never have this case before and it's never been part of the API. I'll use `math.inf` here as the default `null_value` unless people have a strong preference towards NaN.

Reply:
I switched this to use `null_value=math.inf` in: https://github.com/Qiskit/qiskit-terra/pull/9710/commits/b847ebca388be244de98fe6464e5553644d57990